aten/src/ATen/native/cuda/AveragePool2d.cu

```diff
@@ -289,17 +281,19 @@ void avg_pool2d_out_cuda_template(
     inputHeight, inputWidth,
     outputHeight, outputWidth);
 
+  const auto memory_format = input_.suggest_memory_format();
   Tensor input = input_.contiguous(memory_format);
 
   output.resize_({nbatch, nInputPlane, outputHeight, outputWidth});
 
   const int32_t count = safe_downcast<int32_t, int64_t>(output.numel());
   const uint32_t num_threads = std::min(at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, 1024);
-  const uint32_t num_blocks = cuda::ATenCeilDiv<uint32_t>(count, num_threads);
+  const uint32_t num_blocks = cuda::ATenCeilDiv<uint32_t>(count ? count : 1, num_threads);
 
   bool use_divisor = divisor_override.has_value();
   const auto divisor_override_value = use_divisor ? divisor_override.value() : 0; 
 
+  // don't run the kernel if no elements are present.
```

Comment:
Missing a `if(input.numel() != 0)` somewhere?

Reply:
Ah bad actually that comment should be removed it is a remanant of a previous commit.